### DISCUSSION TITLE
How to inform file extension and file type to when uploading File

### DISCUSSION THREAD

**Discussion by RogerioDosSantos** [fastapi/fastapi#8256] [Questions]
I would like to inform the file extension and file type to the OpenAPI.

E.g.:
Content-Disposition: form-data; name="fileName"; filename="*.txt"
Content-Type: text/plain

How can I do it using:

file: UploadFile = File(..., ???)

Thanks in advance

**Comment by tiangolo** [comment]
Right now, as this is not currently tightly integrated, the way to do it would be to modify the generated OpenAPI: https://fastapi.tiangolo.com/tutorial/extending-openapi/

**Comment by tiangolo** [comment]
I'll assume you were able to solve your use case, so I'll close this issue now. But feel free to add more comments or create new issues.

**Comment by Kludex** [comment]
Current link: https://fastapi.tiangolo.com/advanced/extending-openapi/
FastAPI version: 0.60.1
